odl/datasets/ray_transform.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2014-2017 The ODL contributors
+#
+# This file is part of ODL.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Datasets for the ray transform."""
+
+import numpy as np
+from odl.datasets.util import get_data
+from odl.discr import uniform_partition
+from odl.tomo import FanFlatGeometry
+
+
+__all__ = ('walnut_data', 'walnut_geometry',
+           'lotus_root_data', 'lotus_root_geometry')
+
+
+DATA_SUBSET = 'ray_transform'
+
+def walnut_data():
+    """Tomographic X-ray data of a walnut.
+    
+    Notes
+    -----
+    See the article `Tomographic X-ray data of a walnut`_ for further
+    information.
+    
+    See Also
+    --------
+    walnut_geometry
+    
+    References
+    ----------
+    .. Tomographic X-ray data of a walnut: https://arxiv.org/abs/1502.04064
+    """
+    
+    # TODO: Store data in some ODL controlled url
```

Comment:
Well it could also be temporary etc. I'd prefer if we owned the data. Anyway I'll leave it as is for now.